packages/sequence-diagram/src/buildDiagram.ts

```diff
@@ -0,0 +1,180 @@
+import { AppMap, Event } from '@appland/models';
+import { classNameToOpenAPIType } from '@appland/openapi';
+import assert from 'assert';
+import { createHash } from 'crypto';
+import { merge } from './mergeWindow';
+import { selectEvents } from './selectEvents';
+import Specification from './specification';
+import {
+  Action,
+  Actor,
+  Diagram,
+  NodeType,
+  Function,
+  ReturnValue,
+  setParent,
+  Type,
+  ServerRPC,
+} from './types';
+
+const MAX_WINDOW_SIZE = 5;
+
+export default function buildDiagram(
+  appmapFile: string,
+  appmap: AppMap,
+  specification: Specification
+): Diagram {
+  const events = selectEvents(appmap, specification);
+
+  const actors: Actor[] = [];
+  const actorsByCodeObjectId = new Map<string, Actor>();
+
+  const findOrCreateActor = (event: Event): Actor => {
+    const actorCodeObject = specification.isIncludedCodeObject(event.codeObject);
+    assert(actorCodeObject, 'actor code object');
+    const order = specification.priorityOf(actorCodeObject);
+
+    const actorKey = actorCodeObject.fqid;
+
+    let actor = actorsByCodeObjectId.get(actorKey);
+    if (actor) return actor;
+
+    actor = {
+      id: actorCodeObject.fqid,
+      name: actorCodeObject.name,
+      order,
+    } as Actor;
+    actors.push(actor);
+    actorsByCodeObjectId.set(actorKey, actor);
+    return actor;
+  };
+
+  function buildRequest(caller: Event | undefined, callee: Event): Action | undefined {
+    if (callee.httpServerRequest) {
+      assert(callee.route, 'callee.route');
+      assert(callee.httpServerResponse, 'callee.httpServerResponse');
+      return {
+        nodeType: NodeType.ServerRPC,
+        callee: findOrCreateActor(callee),
+        route: callee.route,
+        status: callee.httpServerResponse?.status,
+        digest: callee.hash,
+        subtreeDigest: 'undefined',
+        children: [],
+      } as ServerRPC;
```

Comment:
This cast (and the one below) suppresses type checking; I suspect that's not what you want.